### PR TITLE
SDK-3304: Initial end-to-end tests

### PR DESCRIPTION
**Pull request by NSeydoux**
#187 should be reviewed first.

This adds some end-to-end tests for simple use cases of the library.

### CODE REVIEW

File: e2e/env/.env.example [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Select environment (see @inrupt/internal-test-env for a list of available environments)
+E2E_TEST_ENVIRONMENT="Inrupt Production"
+E2E_TEST_IDP=https://login.inrupt.com
+
+# The following environment variables should be kept in a .env.test.local file
+# Obtain credentials via https://login.inrupt.com/registration.html
+E2E_TEST_CLIENT_ID="<client-id>"
+E2E_TEST_CLIENT_SECRET="<client-secret>"
+E2E_TEST_USER="<username>"
+E2E_TEST_PASSWORD="<password>"
```
Comment (edwardsph) [inline e2e/env/.env.example:10]: Are these actually required in these tests?

**Review by edwardsph — COMMENTED** [review]
LGTM with 1 question

File: e2e/env/.env.example (hunk 1 above)
Comment (NSeydoux) [reply e2e/env/.env.example:10]: Good point, not yet because I didn't add any browser-based tests.

Comment (NSeydoux) [reply e2e/env/.env.example]: Fixed with https://github.com/inrupt/solid-client-errors-js/pull/188/commits/01af408b13ef8ce7f2a742721875e0720bad75df

**Review by edwardsph — APPROVED** [review]
LGTM

**Pull request merged**